### PR TITLE
Video carousel lazy load images

### PR DESCRIPTION
**Pull request by jamesgorrie**
## What does this change?

I've added lazy loading to the video carousel as, on occasion, we can have 5-10 items that bring all their images with - which is a lot.

This container is now on the UK front, so definitely needs trimming, this was the first step. @paperboyo knows exactly how much.
## What is the value of this and can you measure success?

The UK front, and any other lucky front to container this container is slimmer and users have a faster experience.
## Does this affect other platforms - Amp, Apps, etc?

No
## Request for comment

@akash1810 @sammorrisdesign @paperboyo

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/video/video-container.js [hunk 1]
```diff
@@ -67,6 +76,47 @@ define([
         });
     }
 
+    // iniLazyLoadImages is used for when a user scrolls i.e. thin viewport
+    // and fetchImage is used for when the carousel controls are used i.e. fat viewport
+    function initLazyLoadImages() {
```
Comment (jamesgorrie) [inline static/src/javascripts/projects/common/modules/video/video-container.js]: If I could find a way to do `element.on('transform')` we wouldn't have to have the two different ways of doing this.

Or we could have a way of [`element.scrollIntoView({ behaviour: 'smooth' })`](https://developer.mozilla.org/en/docs/Web/API/Element/scrollIntoView#Parameters). 

Writing either of those seemed harder than this.

File: static/src/javascripts/projects/common/modules/video/video-container.js [hunk 2]
```diff
@@ -36,6 +44,7 @@ define([
         $('.video-playlist__item--active').removeClass('video-playlist__item--active');
 
         if (direction === 'next') {
+            fetchImage(containerPos + preloadImageCount);
```
Comment (akash1810) [inline static/src/javascripts/projects/common/modules/video/video-container.js]: worth wrapping this in a promise?

Comment (jamesgorrie) [inline static/src/javascripts/projects/common/modules/video/video-container.js]: What for?

Comment (akash1810) [inline static/src/javascripts/projects/common/modules/video/video-container.js]: to advance only when the image has loaded

Comment (jamesgorrie) [inline static/src/javascripts/projects/common/modules/video/video-container.js]: Na, just do it at the same time... on a slow connection pressing next, then waiting (with no load), to then be scrolled, I would be pissed. Probably best to add a loading image, but I can do that after.

If we went with the next track paradigm, you could definitely do loading, but you would actually be loading the video, which to me would make sense and actually eradicate the need for holding images.

Comment (paperboyo) [inline static/src/javascripts/projects/common/modules/video/video-container.js]: > pressing next, then waiting (with no load), to then be scrolled, I would be pissed
+1

File: static/src/javascripts/projects/common/utils/element-inview.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
+define([
+    'bean',
+    'common/utils/$',
```
Comment (akash1810) [inline static/src/javascripts/projects/common/utils/element-inview.js]: we don't appear to be using `$` `mediator` or `debounce` so they can be removed

**Comment by akash1810** [comment]
👍

File: static/src/javascripts/projects/common/utils/element-inview.js (hunk 3 above)
Comment (jamesgorrie) [inline static/src/javascripts/projects/common/utils/element-inview.js]: We use `debounce` - good catch though.

**Pull request merged**